contracts/Rollup.sol

```diff
@@ -269,6 +282,47 @@ contract Rollup is RollupHelpers {
         );
     }
 
+    function submitBatchWithMM(
+        bytes[] calldata txs,
+        bytes32[] calldata updatedRoots,
+        uint256[2][] calldata aggregatedSignatures,
+        Types.MassMigrationMetaInfo[] calldata MMInfo
+    ) external payable onlyCoordinator {
+        // require(msg.value >= STAKE_AMOUNT, "Not enough stake committed");
+        bytes32[] memory commitments = new bytes32[](updatedRoots.length);
+        bytes32 pubkeyTreeRoot = accountRegistry.root();
+        for (uint256 i = 0; i < updatedRoots.length; i++) {
+            // FIX: This is essentially RollupUtils.MMCommitmentToHash but could not use because of STDeep
+            commitments[i] = keccak256(
+                abi.encode(
+                    updatedRoots[i],
+                    pubkeyTreeRoot,
+                    keccak256(txs[i]),
+                    MMInfo[i].tokenID,
+                    MMInfo[i].amount,
+                    MMInfo[i].withdrawRoot,
+                    MMInfo[i].targetSpokeID,
+                    aggregatedSignatures[i],
+                    Types.Usage.MassMigration
+                )
+            );
+        }
+        Types.Batch memory newBatch = Types.Batch({
+            commitmentRoot: merkleUtils.getMerkleRootFromLeaves(commitments),
```

Comment:
[later] I think we'll have a problem in normal disputeBatch too. The normal disputeBatch should look for previous commitment for the pre state root. If the previous commitment happens to be a mass migration commitment, we can't pass the mass migration commitment to the current disputeBatch since the function arguments have different types. Need to fix this somehow.